tools/pythonpkg/src/python_udf.cpp

```diff
@@ -190,8 +190,8 @@ static scalar_function_t CreateNativeFunction(PyObject *function, PythonExceptio
 					throw NotImplementedException("Exception handling type not implemented");
 				}
 			}
-			python_objects.push_back(py::handle(ret));
-			python_results.push_back(ret);
+			python_objects.push_back(py::reinterpret_steal<py::object>(ret));
```

Comment:
I'm pretty sure this used to be a leak, as `PyObject_CallObject` returns an owned reference, and we were storing them in py::handles, so the refcount would never be decremented.